### PR TITLE
feat: deploy a DHIS2 deployment form

### PR DESCRIPTION
**Pull request by HendrikThePendric**
- Show form fields according to the new deployment strategy
- Fetch fields and values according to the new strategy
- POST data to new endpoints 
- Clean up redundant file
- Add new list view

### CODE REVIEW

**Comment by HendrikThePendric** [comment]
In [bcbeab9](https://github.com/dhis2-sre/im-web-client/pull/362/commits/bcbeab993f3ab25cf4acebb4bb257165946cb9dc) I removed old files, but I decided to keep the folder and component names as-is. We should probably align the component names with the underlying data model, but since that's all still a bit unclear, and because I also need to still refactor the list view in a different PR which is going to touch the same files/folders, I think it's easier to do the renaming later.

File: src/views/instances/new-dhis2/constants.ts [hunk 1]
```diff
@@ -9,3 +11,9 @@ export const INSTALL_REDIS = 'INSTALL_REDIS'
 export const JAVA_OPTS = 'JAVA_OPTS'
 export const OPTIONAL_FIELDS = new Set([JAVA_OPTS])
 export const PUBLIC = 'public'
+
+export const STACK_PRIMARY_PARAMETERS = new Map([
+    ['dhis2-core', new Set(['IMAGE_TAG', 'IMAGE_REPOSITORY'])],
+    ['dhis2-db', new Set(['DATABASE_ID'])],
+    ['pgadmin', new Set(['PGADMIN_USERNAME', 'PGADMIN_PASSWORD'])],
+]) as Dhis2StackPrimaryParameters
```
Comment (kabaros) [inline src/views/instances/new-dhis2/constants.ts:19]: not sure why this needed an assertion (but it's something we can look at later)

File: src/views/instances/new-dhis2/fields/boolean-parameter-checkbox.tsx [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
 import styles from './fields.module.css'
 import { ParameterFieldProps } from './parameter-field'
+import { FC } from 'react'
 
-export const BooleanParameterCheckbox = ({ displayName, parameterName }: ParameterFieldProps) => (
-    <Field type="checkbox" format={formatBool} parse={parseBool} required name={parameterName} label={displayName} component={CheckboxFieldFF} className={styles.parameterCheckbox} />
+export const parseBool = (bool: any): string => (typeof bool === 'boolean' && bool ? 'true' : 'false')
+export const formatBool = (str: string): boolean => str === 'true'
+
+export const BooleanParameterCheckbox: FC<ParameterFieldProps> = ({ displayName, parameterName, stackId }) => (
```
Comment (kabaros) [inline src/views/instances/new-dhis2/fields/boolean-parameter-checkbox.tsx:10]: what version of react is in use? It's just that this way of typing react components has changed, from what I remember, back in react 16...

**Review by kabaros — APPROVED** [review]
it looks good to me - some bits (like the axios hooks) are different from what other projects do, but it's easy to follow and makes sense.

A general comment about naming is that we're usingd `DHIS` as a prefix for things that could have a generic name (for example, `dhisStack` refers to stacks that don't necessarily have to do with DHIS like pgAdmin)

**Review by tonsV2 — APPROVED** [review]

**Pull request merged**